core/src/utils/test/playwright/page/utils/set-content.ts

```diff
@@ -0,0 +1,63 @@
+import type { Page } from '@playwright/test';
+
+/**
+ * Overwrites the default Playwright page.setContent method.
+ *
+ * Navigates to a blank page, sets the content, and waits for the
+ * Stencil components to be hydrated before proceeding with the test.
+ *
+ * @param page The Playwright page object.
+ * @param html The HTML content to set on the page.
+ */
+export const setContent = async (page: Page, html: string) => {
+  if (page.isClosed()) {
+    throw new Error('setContent unavailable: page is already closed');
+  }
+
+  if (typeof html !== 'string') {
+    throw new Error('invalid html: must be a string');
+  }
```

Comment:
Is this needed? TS should catch it for us.

Reply:
You're right 👍  I copied from the Stencil Puppeteer implementation, but I don't see any reason/value of this check.